app/uk/gov/hmrc/apinotificationpull/connectors/ApiNotificationQueueConnector.scala

```diff
@@ -43,6 +42,8 @@ class ApiNotificationQueueConnector @Inject()(config: ServicesConfig, http: Http
     logger.debug(s"Getting notification by id using url: $url")
     http.GET[HttpResponse](url)
       .map { r =>
+        if(r.status == 404){throw new NotFoundException("")}
```

Comment:
REmove this and change the recover bit to upstream error response and check status 404 on that one. We shouldn't use NotFoundException anymore

Reply:
Although if this works fine as is and is creating more issues by introducing upstream error then leave it for time being